src/move/research/prover-examples/sources/main.move

```diff
@@ -0,0 +1,53 @@
+module prover_examples::main {
+
+    use std::signer;
+
+    const MAX_VALUE: u8 = 100;
+
+    /// The provided value is too high.
+    const E_VALUE_TOO_HIGH: u64 = 0;
+
+    struct PhantomKeyStruct<phantom T> has key {
+        account_address: address,
+        value: u8
+    }
+
+    spec PhantomKeyStruct {
+        invariant value <= MAX_VALUE;
+    }
+
+    invariant<T> forall account_address: address where exists<PhantomKeyStruct<T>>(
+        account_address
+    ): global<PhantomKeyStruct<T>>(account_address).account_address == account_address;
+
+    public fun move_to_phantom_key_struct<T>(account: &signer, value: u8) {
+        assert!(value <= MAX_VALUE, E_VALUE_TOO_HIGH);
```

Comment:
Shouldn't we remove this since we have the spec ?

I would guess that this costs gas. If the prover proves this cannot happen, then let's get rid of the (now) useless asserts.

Reply:
> Shouldn't we remove this since we have the spec ?

Try removing it and running the prover and see what happens ;)